### PR TITLE
Add line function to cycle-canvas

### PR DESCRIPTION
**Pull request by zestree**
I implemented the line function with the following api:
line({
    points: [
      {x:0, y: 0},
      {x: 20, y: 20},
      {x: 40, y: 20},
      {x: 50, y: 30}
    ],
    style: {
      lineWidth: 2,
      strokeStyle: 'red',
      lineCap: 'butt',
      lineJoin: 'round',
      lineDash: [5, 2]
    }
})

I changed property `draw: []` (from rect and text) to `style: {}` for easier assignment of variable to context.
Please let me know if you have any thoughts or comments.
Thanks

### CODE REVIEW

**Comment by Widdershin** [comment]
Hi edwin. `draw` was an array because the order of operations (fill then stroke vs sroke then fill) changes the result.

Perhaps we can just add a flag to style to allow changing the order.

I'm off to catch a plane in a few hours, and will be travelling for the next 24 hours or so, so I won't be able to review and merge your pull request for a while.

**Comment by zestree** [comment]
Hi Nick, I see. I'm wondering if we should just keep using 'draw' for rect
and text, and use 'style' for line. Because they behave slightly
differently, as line just need the property value and doesn't care about
order

**Comment by Widdershin** [comment]
>  I'm wondering if we should just keep using 'draw' for rect and text, and use 'style' for line. Because they behave slightly differently, as line just need the property value and doesn't care about order

That seems like a good approach for now. 😄 

I've tried this out and I think I found a bug:

``` js
import {line} from '../../src/canvas-driver';

import {Observable} from 'rx';

function view () {
  return (
    line({
      points: [
        {x: 100, y: 100},
        {x: 100, y: 200},
        {x: 200, y: 100},
        {x: 100, y: 100}
      ],

      style: {}
    })
  );
}

export default function App ({Canvas, Keys, Animation}) {
  return {
    Canvas: Observable.just(view())
  };
}
```

I would expect that to produce a triangle, but this is the result:

screen shot 2016-04-27 at 8 29 33 am

It looks like it's always starting at 0, 0. I would expect that the first point is the starting point.

File: src/canvas-driver.js [hunk 1]
```diff
@@ -7,79 +7,114 @@ function renderElement (context, element, parent) {
...
+
+  if (element.kind === 'line') {
+    drawLine(context, element, origin);
+  }
+
+  element.children && element.children.forEach(child => renderElement(context, child, element))
+}
+
+function drawLine(context, element, origin) {
+  context.lineWidth = element.style.lineWidth || 1;
```
Comment (Widdershin) [inline src/canvas-driver.js:35]: If the user calls `line()` without providing a style object, they get this error: `Uncaught TypeError: Cannot read property 'lineWidth' of undefined`.

Could we use `defaults` in the `line` helper, in the same way `text` is done?

**Comment by zestree** [comment]
oops my bad, just pushed some changes. cheers

**Comment by Widdershin** [comment]
Awesome, this looks great! Thanks so much for your contribution. Are you keen to take on some more functionality? Perhaps circles with arc?

**Comment by zestree** [comment]
Hi Nick, no worries. Sure I'll take a look at circles when I have some free time.

**Comment by Widdershin** [comment]
@edwin-nz: awesome. In the meantime, how would you feel about being added as a collaborator to this repo? It would mean you could work on branches in this repo, which means we can collaborate over pull requests more easily.

**Comment by zestree** [comment]
Hey thanks for the suggestion. I do like the idea, but it just happened
that I'm now dealing with a problem with my house, so am fully occupied and
will not be able to contribute until the issue is resolved...

**Comment by Widdershin** [comment]
Hi Edwin, that's perfectly okay. I hope you sort out your problem without
too much pain.

In the meantime I've added you as a collaborator. Please don't worry if you
can't contribute regularly (or at all), I expect that you look after
yourself first of all :). Thanks very much for your contribution so far.
Hey thanks for the suggestion. I do like the idea, but it just happened
that I'm now dealing with a problem with my house, so am fully occupied and
will not be able to contribute until the issue is resolved...

**Pull request merged**